src/commands/map.test.ts

```diff
@@ -0,0 +1,450 @@
+          })
+        ).rejects.toThrow(
+          'Missing provider name. Please provide it as first argument.'
+        );
+      });
+
+      it('throws when provider name is invalid', async () => {
+        jest.mocked(exists).mockResolvedValueOnce(true);
+        jest
+          .mocked(readFile)
+          .mockResolvedValueOnce(profileSource(profileScope, profileName));
+        await expect(
+          instance.execute({
+            logger,
+            userError,
+            flags: {},
+            args: {
+              providerName: '!_0%L',
+              profileId: `${profileScope}.${profileName}`,
+            },
+          })
+        ).rejects.toThrow('Invalid provider name');
+      });
+
+      it('throws when provider file does not exist', async () => {
+        jest.mocked(exists).mockResolvedValueOnce(true);
+        jest
+          .mocked(readFile)
+          .mockResolvedValueOnce(profileSource(profileScope, profileName));
+        await expect(
+          instance.execute({
+            logger,
+            userError,
+            flags: {},
+            args: {
+              providerName: 'test',
+              profileId: `${profileScope}.${profileName}`,
+            },
+          })
+        ).rejects.toThrow(
+          'Provider test does not exist. Make sure to run "sf prepare" before running this command.'
+        );
+      });
+
+      it('throws when reading of file fails', async () => {
+        jest.mocked(exists).mockResolvedValueOnce(true);
+        jest
+          .mocked(readFile)
+          .mockRejectedValueOnce(new Error('File read error'));
+        await expect(
+          instance.execute({
+            logger,
+            userError,
+            flags: {},
+            args: {
+              providerName: 'test',
+              profileId: `${profileScope}.${profileName}`,
+            },
+          })
+        ).rejects.toThrow('File read error');
+      });
+
+      it('throws when provider is not valid JSON', async () => {
+        jest.mocked(exists).mockResolvedValueOnce(true);
+        jest.mocked(readFile).mockResolvedValueOnce('file content');
+        await expect(
+          instance.execute({
+            logger,
+            userError,
+            flags: {},
+            args: {
+              providerName: 'test',
+              profileId: `${profileScope}.${profileName}`,
+            },
+          })
+        ).rejects.toThrow(``);
+      });
+
+      it('throws when provider is not Provider JSON', async () => {
+        jest.mocked(exists).mockResolvedValueOnce(true);
+        jest.mocked(readFile).mockResolvedValueOnce('{"test": 1}');
+        await expect(
+          instance.execute({
+            logger,
+            userError,
+            flags: {},
+            args: {
+              providerName: 'test',
+              profileId: `${profileScope}.${profileName}`,
+            },
+          })
+        ).rejects.toThrow(``);
```

Comment:
question: Is it correct that it throws empty string as error message?

Reply:
Nope, I will add expected errors